### PR TITLE
Config package implementation

### PR DESCRIPTION
**Pull request by maxence-charriere**
Hey guys, this is a package to handle config and command line args in a
simple way.
A little review would be awesome.
I d like to use it in our workers.

@pavitrabhalla
@ravidharawat

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	Foo     string `json:"foo"     help:"A foo"        env:"EXAMPLE_FOO"`
+	Bar     int    `json:"bar"     help:"A bar"`
+	Verbose bool   `json:"verbose" help:"Verbose mode"`
+}
+
+func main() {
+	os.Setenv("EXAMPLE_FOO", "Hello")
+
+	config.ConfigName = "config.json"
+	// config.CreateIfNotExists = true
```
Comment (pavitrabhalla) [inline README.md]: Whats this comment for?

Comment (maxence-charriere) [inline README.md]: just commented the example to not create the config file if it not exists.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+	cfg := conf{
+		Foo: "while",
+		Bar: 42,
+	}
+
+	cmds := config.Load(&cfg)
+	fmt.Printf("cmds: %+v\n", cmds)
```
Comment (pavitrabhalla) [inline README.md]: Can you change the print `cmds` to `Commands` for better understanding?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	fmt.Printf("cfg: %+v\n", cfg)
```
Comment (pavitrabhalla) [inline README.md]: Also, maybe print `Configuration:` instead of `cfg:`

File: file.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
+package config
+
+import (
+	"encoding/json"
+	"os"
+)
+
+type fileLoader struct {
+	name              string
```
Comment (pavitrabhalla) [inline file.go]: Does this take a fully qualified file path, or a relative path, or both?

Comment (maxence-charriere) [inline file.go]: both

Comment (maxence-charriere) [inline file.go]: it uses the standard behavior for open file

**Comment by maxence-charriere** [comment]
@pavitrabhalla 
modified :)

**Comment by pavitrabhalla** [comment]
👍  Lets get this merged. :-)

**Pull request merged**